client/src/utils/Geometry.ts

```diff
@@ -61,8 +92,39 @@ export function calculateLinesIntersection(line1: Line, line2: Line): Dot | null
     return intersection;
 }
 
-export function eucledianDistance(dot1: Dot, dot2: Dot): number {
-    const dx = dot1.x - dot2.x;
-    const dy = dot1.y - dot2.y;
-    return Math.sqrt(dx*dx + dy*dy);
+/** Caution: Always returns an angle smaller (or equal to) 180 degrees */
+export function findAngleBetweenTwoVectors(vector1: Dot, vector2: Dot): number {
+    const dotProduct = calculateDotProduct(vector1, vector2);
+    const magnitude1 = calculateVectorMagnitude(vector1);
+    const magnitude2 = calculateVectorMagnitude(vector2);
+    const magnitudeProduct = magnitude1 * magnitude2;
+
+    return radiansToDegrees(Math.acos(dotProduct/magnitudeProduct));
+}
+
+export function findAngleFromReferenceLine(reference: Line, line: Line): number {
+    let referenceVector = normalizeVector(lineToVector(reference));
+    let lineVector = normalizeVector(lineToVector(line));
+
+    const dotProduct = calculateDotProduct(referenceVector, lineVector);    
+    const crossProduct = calculateCrossProduct(referenceVector, lineVector);
+
+    let angle = radiansToDegrees(Math.atan2(crossProduct, dotProduct));
+
+    return (angle > 0) ? (360 - angle) : (-angle);
+}
+
+export function rotatePointClockwise(point: Dot, angle: number, referencePoint: Dot = ORIGIN_POINT): Dot {
```

Comment:
The `referencePoint` is always provided, so its default `ORIGIN_POINT` can be safely removed.
This default parameter because it makes it easy to forget to provide the `referencePoint` parameter.

I think that ORIGIN_POINT will never be the correct `referencePoint` (the origin point, that I don't fully understand, represents upper left corner).

I would prefer forcing a user of the function to provide `referencePoint` they want, thus removing `ORIGIN_POINT` default.